### ISSUE TITLE
Assertion violation at mcsat/trail.c:120

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#182]
Hi, for the following formula,
~~~~
(set-logic QF_NRA)
(declare-const r0 Real)
(declare-const r2 Real)
(assert (< r2 0.0 r0 r2 r0))
(push 1)
(check-sat)
(check-sat)
(pop 1)
~~~~

yices2 throws an assertion violation 
~~~~
unsat
unsat
yices_smt2: mcsat/trail.c:120: trail_pop_base_level: Assertion `trail->decision_level == trail->decision_level_base' failed.
~~~~

OS: Unbuntu 16.04
Commit: 65cdbeeba29cc

**Comment by dddejan** [comment]
Thanks for report. 

Same issue as #180.

**Comment by dddejan** [comment]
Fixed with fix for #180
